engine/src/flutter/impeller/renderer/backend/gles/pipeline_library_gles.cc

```diff
@@ -209,14 +210,22 @@ std::shared_ptr<PipelineGLES> PipelineLibraryGLES::CreatePipeline(
 
   const auto has_cached_program = !!cached_program;
 
-  auto pipeline = std::shared_ptr<PipelineGLES>(new PipelineGLES(
-      reactor,       //
-      weak_library,  //
-      desc,          //
-      has_cached_program
-          ? std::move(cached_program)
-          : std::make_shared<UniqueHandleGLES>(UniqueHandleGLES::MakeUntracked(
-                reactor, HandleType::kProgram))));
+  std::shared_ptr<UniqueHandleGLES> program_handle = nullptr;
+  if (has_cached_program) {
+    program_handle = std::move(cached_program);
+  } else {
+    program_handle = threadsafe ? std::make_shared<UniqueHandleGLES>(
+                                      reactor, HandleType::kProgram)
+                                : std::make_shared<UniqueHandleGLES>(
+                                      UniqueHandleGLES::MakeUntracked(
+                                          reactor, HandleType::kProgram));
+  }
+
+  auto pipeline = std::shared_ptr<PipelineGLES>(
+      new PipelineGLES(reactor,       //
+                       weak_library,  //
+                       desc,          //
+                       std::move(program_handle)));
```

Comment:
```suggestion
  auto pipeline = std::make_shared<PipelineGLES>(reactor,       //
                       weak_library,  //
                       desc,          //
                       std::move(program_handle));
```
How about changing like above?

Reply:
These are flutter sources. For code management purposes, I didn't changed.

Reply:
I see.